core/templates/dev/head/pages/exploration-player-page/suggestion-modal-for-learner-local-view/suggestion-modal-for-exploration-player.service.ts

```diff
@@ -61,7 +61,14 @@ angular.module('oppia').factory('SuggestionModalForExplorationPlayerService', [
             $timeout(function() {
               $scope.showEditor = true;
             }, 500);
-
+            $scope.detectChange = function(e) {
+              if ($($scope.suggestionData.suggestionHtml).html() ===
```

Comment:
@donosco98 Any chance you could find a more "angular"-ish way to do this? We typically try to avoid using jquery functions, so I'm trying to understand ... what's wrong with just comparing $scope.suggestionData.suggestionHtml and $scope.originalHtml directly? Can you provide a concrete example where the two methods give different results?

Reply:
![console](https://user-images.githubusercontent.com/31371571/69527917-4c200f00-0f93-11ea-9c24-e1ae137394e3.png)
@seanlip I have logged all the things which happen before loading the data in ckeditor i.e initial state
-the first and second log is ` $scope.originalHtml ` and `$scope.suggestionData.suggestionHtml` before ckeditor and wrap function is applied.
Till here there is no problem.
-The last 2 logs are in ckeditor before loading and after `wrapComponents` and `setData`
Here `ck.setData` is for some reason is removing the image component html completely.
I think this is were the problem is. 
Any suggestions?

Reply:
Also ,`ck.setData`works fine if I remove the `wrapComponent` in line 196 before passing into `setData` 
![aferer](https://user-images.githubusercontent.com/31371571/69528632-a53c7280-0f94-11ea-9370-54634f38a94d.png)
These logs come if we dont apply `wrapComponents` before passing to `ck.setData`

Reply:
@seanlip , the jquery code works because it strips away all the HTML tags and compares only the content so It works as expected

Reply:
Thanks @donosco98! I'm not sure what's the cause of this -- you'll need to debug. But this is a good start. See if you can figure out why the component is getting removed. (Stripping HTML tags arbitrarily is probably not the way to go -- that seems kinda hacky.)

Reply:
Yeah I'll try that!

Reply:
@seanlip,I've fixed the ckeditor problem by disabling content filtering in ckeditor!